edi-module/edi_translator.bal

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+
+type EDIContext record {|
+    EDISchema schema;
+    string[] ediText = [];
+    int rawIndex = 0;
+|};
+
+# Reads the given EDI text according to the provided schema.
+#
+# + ediText - EDI text to be read
+# + schema - Schema of the EDI text
+# + return - JSON variable containing EDI data. Error if the reading fails.
+public isolated function read(string ediText, EDISchema schema) returns json|Error {
+    EDIContext context = {schema};
+    EDIUnitSchema[] currentMapping = context.schema.segments;
+    context.ediText = splitSegments(ediText, context.schema.delimiters.segment);
+    EDISegmentGroup rootGroup = check readSegmentGroup(currentMapping, context, true);
+    return rootGroup.toJson();
+}
+
+# Writes the given JSON varibale into a EDI text according to the provided schema.
+#
+# + msg - JSON value to be written into EDI
+# + schema - Schema of the EDI text
+# + return - EDI text containing the data provided in the JSON variable. Error if the reading fails.
+public isolated function write(json msg, EDISchema schema) returns string|Error {
+    EDIContext context = {schema};
+    if !(msg is map<json>) {
```

Comment:
Make this the first statement of the function, as there is no point in creating `EDIContext` if this error occurs.